docker/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Base image
+FROM ghcr.io/scientificcomputing/fenics:2023-03-01
+
+WORKDIR /tmp/
+RUN mkdir -p /tmp/cbc.block
+COPY . /tmp/cbc.block/
+
+ARG HAZMATH_VERSION=v1.0.1
+ARG PYAMG_VERSION=v4.2.3
+
+ENV SUITESPARSE_DIR=/usr/local/petsc/linux-gnu-real-32/
+ENV PYTHONPATH=""
+ENV HAZMATH_VERSION=$HAZMATH_VERSION
+ENV PYAMG_VERSION=$PYAMG_VERSION
+
+RUN git clone --branch=${PYAMG_VERSION} --single-branch --depth=1 https://github.com/pyamg/pyamg.git && \
+    python3 -m pip install pyamg
+
+# Haznics
+RUN apt-get update && apt-get install -y swig && apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /var/tmp/*
+RUN git clone --branch=${HAZMATH_VERSION} --single-branch --depth=1 https://github.com/HAZmathTeam/hazmath && \
+    cd hazmath && \ 
+    make config shared=yes suitesparse=yes lapack=yes haznics=yes swig=yes . && \
+    make install
+
+ENV PYTHONPATH /tmp/hazmath/swig_files:${PYTHONPATH}
+
+# cbc.block
+COPY . /tmp/cbc.block/
+RUN python3 -m pip install cbc.block/[all]
```

Comment:
It might be nice to add `-e` so that the code is not copied away from /tmp/cbc.block. That way, it is possible to bind-mount it from the outside for a "live" environment.